src/main/java/bio/terra/cli/model/TerraUser.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.cli.model;
+
+import bio.terra.cli.utils.AuthenticationUtils;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.ServiceAccountCredentials;
+import com.google.auth.oauth2.UserCredentials;
+import java.util.Date;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This POJO class represents a Terra identity context, which includes all related credentials (e.g.
+ * user, pet SA).
+ */
+public class TerraUser {
+  private static final Logger logger = LoggerFactory.getLogger(TerraUser.class);
+
+  public String cliGeneratedUserKey;
+  public String terraUserId;
+  public String terraUserName;
+  @JsonIgnore public UserCredentials userCredentials;
+  @JsonIgnore public ServiceAccountCredentials petSACredentials;
```

Comment:
I think for non-Google credentials, it would be better to have an object that holds user + sa credentials and implements some common interface. Then we could use variable names to hold one for each cloud. So prefer to hold off on this change here specifically.